review/views.py

```diff
@@ -55,12 +64,22 @@ class BookDetailView(DetailView):
     review_form = ReviewForm()
     comment_form = CommentForm()
 
-    def book_detail_view(request, primary_key):
-        book = get_object_or_404(Book, pk=primary_key)
-        return render(request, 'books/book_detail.html', context={'book': book})
+    def get_success_url(self):
+        return reverse('book-detail', kwargs={'pk': self.object.pk})
+
+    # def book_detail_view(self, request, primary_key):
+    #     book = get_object_or_404(Book, pk=primary_key)
+    #     bookmark = BookMark.objects.filter(user=self.request.user.id, book=book.id)
+    #     print(bookmark)
+    #     return render(request, 'books/book_detail.html', context={'book': book, 'bookmark': bookmark})
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
+        book = self.get_object()
+        context['book'] = book
+        bookmark = BookMark.objects.filter(user=self.request.user.id, book=book.id).first()
```

Comment:
filter ko có thì sao mà .first được, nên xử lý an toàn bằng việc check trước nha